docs/superpowers/specs/2026-03-14-tier3-context-efficiency-design.md

```
@@ -0,0 +1,143 @@
+
+`Summarizer` struct holds a provider reference.
+
+**`MaybeSummarize(ctx, messages) ([]provider.Message, error)`:**
+
+1. If `len(messages) < summarizeThreshold` (default 30), return as-is
+2. Split: `first` (message[0]), `middle` (to compress), `recent` (last 10)
+3. Call LLM with summary prompt against middle section
+4. Return `[first, summaryMessage, recent...]`
+
+**Summary prompt:**
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)